.circleci/config.yml

```diff
@@ -152,6 +160,10 @@ workflows:
           filters:
             tags:
               only: /^v.*/
+
+      - deploy_new_chain_pr_only:
+          requires:
+            - tests
```

Comment:
I believe it requires e2e because only e2e publishes the docker image.

Reply:
oh, that is strange. The tests are passing while I specify the commit based docker tag.
Could it be that Boyar deployed something else in it's absence after failing silently to find the correct image?

Can you double check that ?

Reply:
Usually the `build` or `tests` steps which produces the binary should finish with exporting a working and tested image. Why the E2E from all things?

Reply:
Because we need run e2e inside docker and we need to build the image anyway. After the image passes e2e, we can export it.

Tests run separately for quick feedback.

The explanation why your code still works is simple: boyar can't pull the image before it is built, so it retries and retries until it succeeds (meaning e2e finished and uploaded the image).